### PR TITLE
Multiple Matmul Scheduler: schedule mma instruction

### PR DESCRIPTION
**Pull request by jacobhinkle**
This is the latest in a series of PRs. It schedules `mma_result`.
    
Stacked on #2992.

### CODE REVIEW

**Comment by jacobhinkle** [comment]
!build

File: csrc/scheduler/mma_utils.cpp [hunk 1]
```diff
@@ -281,6 +281,143 @@ void scheduleWarpTileWithReduction(TensorView* tv, MatMulTileOptions tile) {
   // [Kwo Mwo Nwo Mw Nw Mi Ni Ki]
 }
 
+void scheduleWarpTile(
```
Comment (jacobhinkle) [inline csrc/scheduler/mma_utils.cpp:284]: This adapts `scheduleWarpTileWithReduction` to accept a vector of dim roles, which might not be M, N, K. I intend to eventually switch to using this in `schedulePrologue` as well in order to support different dimension ordering like N, K, M.

File: csrc/scheduler/mma_utils.cpp [hunk 2]
```diff
@@ -281,6 +281,143 @@ void scheduleWarpTileWithReduction(TensorView* tv, MatMulTileOptions tile) {
...
+
+  // Do not split K dimension of CTA tile into multiple warp tiles
+  NVF_CHECK(
+      cta_tile.k == warp_tile.k,
+      "CTA tile and warp tile must have same K dimension");
+
+  // Find _inner_ tile dimensions present in tv
+  std::vector<int64_t> inner_dims;
+  std::vector<int64_t> required_inner_dim_sizes;
+  int64_t m_dim = -1, n_dim = -1, k_dim = -1;
```
Comment (zasdfgbnm) [inline csrc/scheduler/mma_utils.cpp:301]: Please include your improvements in https://github.com/NVIDIA/Fuser/pull/2992, such as https://github.com/NVIDIA/Fuser/pull/2992/commits/5c19d4edee3be8ec9eedeaf0b53f97e62f719f21.

Comment (jacobhinkle) [reply csrc/scheduler/mma_utils.cpp:301]: Ah of course. I'll re-request a review when it's ready again.

File: csrc/scheduler/multi_matmul.cpp [hunk 3]
```diff
@@ -1006,6 +1006,37 @@ class MultipleMatmulScheduler {
     scheduleBranch(bs_, bcw_smems_, params_->supported_vec_size.b);
   }
 
+  // Schedule acr and bcr
+  void scheduleLdMatrixLoads() {
```
Comment (zasdfgbnm) [inline csrc/scheduler/multi_matmul.cpp:1010]: Hmm... Looks like this and `scheduleWarpTile` is still not used?

File: csrc/scheduler/multi_matmul.cpp [hunk 4]
```diff
@@ -1050,7 +1081,154 @@ class MultipleMatmulScheduler {
...
   void scheduleMmaResults() {
-    NVF_THROW("scheduleMmaResults is not yet implemented");
+    auto all_merged_roles = blockTileTensors(mma_results_);
+    for (size_t i : c10::irange(mma_results_.size())) {
+      TensorView*& mma_result = mma_results_[i];
+      std::vector<MatmulDimRole>& merged_roles = all_merged_roles[i];
+
+      // do split-K rFactor to define splitk_sum and smem_epilogue
+      if (params_->splitk_factor != 1) {
+        TensorView* splitk_sum = mma_result->rFactor({-4, -1});
```
Comment (zasdfgbnm) [inline csrc/scheduler/multi_matmul.cpp:1091]: Who did the split for split-k? Is it `blockTileTensors`? How about adding this information as a comment in code?

**Review by zasdfgbnm — APPROVED** [review]

File: csrc/scheduler/multi_matmul.cpp (hunk 3 above)
Comment (jacobhinkle) [reply csrc/scheduler/multi_matmul.cpp:1010]: You're right again. Removing it again. Sorry about that.

**Pull request merged**